modules/kinesis/src/main/scala/com/snowplowanalytics/snowplow/enrich/kinesis/Sink.scala

```diff
@@ -139,10 +145,15 @@ object Sink {
         wasSuccessful = _.isSuccessful,
         policy = retryPolicy,
         onFailure = (result, retryDetails) =>
-          Logger[F].warn(s"Writing to shard ${result.getShardId()} failed after ${retryDetails.retriesSoFar} retry"),
+          Logger[F]
+            .warn(
+              s"Writing to shard ${result.getShardId()} failed after ${retryDetails.retriesSoFar} retries. Errors : [${getErrorMessages(result)}]"
```

Comment:
just a nit, since we let KPL retry forever, `retryDetails.retriesSoFar` will always be 0 until KPL retries Long.MaxValue times

Reply:
I should make it clear what is KCL retry and what is cats-retry retry

Reply:
> since we let KPL retry forever, retryDetails.retriesSoFar will be 0 until KPL retries
I wonder if we can be 100% sure that KPL won't fail for some unexpected reason, so I would still let the number of cats-retry retries, what do you think ?

Please see https://github.com/snowplow/enrich/pull/552/commits/6794e4fb8c84a79eb9db7ff47bf8adaf3daaa1b4

Reply:
>  KPL won't fail for some unexpected reason

I'd prefer to protect ourselves against that so I'd let cats-retry retry forever as well and report both retry numbers in log message